vm/src/stdlib/collections.rs

```diff
@@ -3,24 +3,26 @@ pub(crate) use _collections::make_module;
 #[pymodule]
 mod _collections {
     use crate::builtins::pytype::PyTypeRef;
+    use crate::builtins::PyInt;
     use crate::common::lock::{PyRwLock, PyRwLockReadGuard, PyRwLockWriteGuard};
-    use crate::function::OptionalArg;
+    use crate::function::{FuncArgs, OptionalArg};
     use crate::slots::{Comparable, Hashable, Iterable, PyComparisonOp, PyIter, Unhashable};
     use crate::vm::ReprGuard;
     use crate::VirtualMachine;
     use crate::{sequence, sliceable};
     use crate::{PyComparisonValue, PyIterable, PyObjectRef, PyRef, PyResult, PyValue, StaticType};
     use itertools::Itertools;
+    use num_traits::ToPrimitive;
     use std::collections::VecDeque;
 
     use crossbeam_utils::atomic::AtomicCell;
 
     #[pyattr]
     #[pyclass(name = "deque")]
-    #[derive(Debug)]
+    #[derive(Debug, Default)]
     struct PyDeque {
         deque: PyRwLock<VecDeque<PyObjectRef>>,
-        maxlen: Option<usize>,
+        maxlen: AtomicCell<Option<usize>>,
```

Comment:
Missed the fact that its mutated in the last review, unfortunately.

Reply:
I'd have to look into it because I'm not sure how to do it now. Wouldn't that require getting a lock for reading it, too?

If you have a way to do it now, just push to get the branch directly

Reply:
I pushed a commit. I also fixed a bug that `maxlen=None` is ignored

Reply:
Good catch on `maxlen`. That's a pretty nifty trick there. I'll keep it in my in case this pops up someplace else.